meta-lmp-bsp/dynamic-layers/freescale-layer/recipes-bsp/imx-mkimage/imx-boot_%.bbappend

```diff
@@ -34,4 +40,8 @@ do_compile_prepend_mx8m() {
 do_deploy_append() {
     # Also create imx-boot link with the machine name
     ln -sf ${BOOT_CONFIG_MACHINE}-${IMAGE_IMXBOOT_TARGET} ${DEPLOYDIR}/${BOOT_NAME}-${MACHINE}
+    SECONDARY_OFFSET=$(printf "%d" $(grep -e CONFIG_SECONDARY_BOOT_SECTOR_OFFSET ${DEPLOY_DIR_IMAGE}/u-boot-fio-config | sed 's/=/ /g' | awk '{print $2}'))
+    if [ "${SECONDARY_OFFSET}" != "0" ]; then
+        install -m 0644 ${S}/sit.bin ${DEPLOYDIR}/sit-${MACHINE}.bin
```

Comment:
Having another look here (had the impression this was another do_compile piece initially), you can just do an if [ -f ${S}/sit.bin ] and install if the file is avalable. No need for fetching the offset.

Reply:
agree with this ^^